src/lib.rs

```diff
@@ -615,6 +614,22 @@ impl<K: Hash + Eq + PartialOrd, V: PartialOrd, S: HashState> PartialOrd for Link
     fn partial_cmp(&self, other: &LinkedHashMap<K, V, S>) -> Option<Ordering> {
         iter::order::partial_cmp(self.iter(), other.iter())
     }
+
+    fn lt(&self, other: &LinkedHashMap<K, V, S>) -> bool {
+        iter::order::lt(self.iter(), other.iter())
+    }
+
+    fn le(&self, other: &LinkedHashMap<K, V, S>) -> bool {
+        iter::order::le(self.iter(), other.iter())
+    }
+
+    fn ge(&self, other: &LinkedHashMap<K, V, S>) -> bool {
+        iter::order::ge(self.iter(), other.iter())
```

Comment:
It's not clear to me that ge/le/ne have different perf characteristics; surely they must all short-circuit at the same place?